src/speed-index.js

```diff
@@ -59,30 +89,26 @@ function calculateFrameSimilarity(frame, target) {
 	return diff.ssim;
 }
 
-function calculatePerceptualProgress(frames) {
+function calculatePerceptualProgress(frames, opts) {
+	const initial = frames[0];
 	const target = frames[frames.length - 1];
-
-	// Calculate frames simliarity between each frames and the final
-	const framesSimilarity = frames
-		.map(frame => calculateFrameSimilarity(frame, target));
-
-	// Get the min frame similarity value
-	const minPreceptualProgress = framesSimilarity
-		.reduce((min, progress) => Math.min(min, progress), Infinity);
-
-	// Remap the values from [minPreceptualProgress, 1], to [0, 100] interval
-	// to be consistent with the standard visual progress
-	const normalizedSimilarity = framesSimilarity
-		.map(progress => {
-			if (progress === minPreceptualProgress) { // Images are the same
-				return 0;
+	const initialSimilarity = calculateFrameSimilarity(initial, target);
+
+	calculateProgressBetweenFrames(
+		frames,
+		0,
+		frames.length - 1,
+		opts && opts.fast ? 5 : 0,
```

Comment:
set 5 as a constant above? 

reading this line it's hard to understand what "5" as a "threshold" even means.